src/foam/util/AddressUtil.java

```diff
@@ -57,4 +66,73 @@
     return new String[] { streetNumber, streetName };
   }
 
+  public static String normalizeRegion(X x, String country, String regionCode) {
+    String normalizedRegion = country + "-" + regionCode;
+    DAO regionDAO = (DAO) x.get("regionDAO");
+    Region found = (Region) regionDAO.find(regionCode);
+    if ( found != null )
+      return found.getCode();
+
+    ArraySink sink = (ArraySink) regionDAO.where(AND(
+      EQ(Region.COUNTRY_ID, country),
+      OR(
+        EQ(Region.ISO_CODE, regionCode),
+        CONTAINS_IC(Region.NAME, regionCode)
```

Comment:
May use starts_with_ic?

Reply:
STARTS_WITH_IC does not have a Java implementation, only a javascript implementation.

Reply:
Looks like the MLang.java implementation of STARTS_WITH_IC does not exist. I'll try to add it

Reply:
Updated both CONTAINS_IC to STARTS_WITH_IC.